explainer.md

```diff
@@ -0,0 +1,87 @@
+
+The table below shows that currently only same-document atomic navigations can be customized.
+
+|   | Same-Document  |  Cross-Document Same-Origin | Cross-Origin  |
+|---|---|---|---|
+| Swipe Navigation | ❌ | ❌ | ❌ |
+| Atomic Navigation  | ✅ | ❌ | ❌ |
+
+Ideally all same-origin navigations should be customizable but the following API gaps remain to enable that:
+
+* Gesture API: Most platforms have a swipe gesture which is used by the UA to trigger a back/forward navigation. On Android/iOS its swiping from the screen edge; on Mac, its multi-finger swipe. Site authors currently can not consistently intercept these events. Customizing swipe navigations for both same-document and cross-document navigations requires this[^1].
+
+* Persisting Visual State: The transition requires showing visual state from both before/after the navigation. Site authors can do this for same-document navigations but not cross-document. [ViewTransition](https://github.com/WICG/view-transitions/blob/main/explainer.md#cross-document-same-origin-transitions) addresses this gap.
+
+The exact details for the above APIs is out of scope for this proposal. These are meant to show that there are cases which are currently not customizable by site authors but will be going forward. The API proposals here should work well if/when one of these cases becomes customizable.
+
+# Problem Statement
+There is currently no way for a site to indicate that it has a custom transition which should override the default UA transition. For example, if the UA adds a visual transition for an atomic navigations (like clicking the back button) which the site has also customized.
+
+## Same-Document Navigations
+For same-document navigations the UA can not detect whether a site will execute a visual transition for a navigation. Also, while the site can not currently customize the swipe gesture, it may still do a visual transition when the navigation is eventually committed. Adding UA transitions for these navigations has a compat risk of "double transitions". This [site](https://darkened-relieved-azimuth.glitch.me) shows the issue on Chrome/Safari iOS and Safari on Mac. The site does a cross-fade when the user goes back:
+
+* If the user navigates using the back button in the UI, a cross-fade defined by the site executes and is the only transition.
+
+* If the user navigates using a swipe from the edge (iOS) or 2 finger swipe (mac), the UA does a transition showing content of the back entry. The Document then receives a `popState` event for the navigation and performs another visual transition.
+
+## Cross-Document Navigations
+This is not a problem for cross-document navigations since customization requires UA primitives. The UA can use this to detect if there are custom transitions and give them precedence. For example:
```

Comment:
Cross-document navigations don't have the same problems as same-document navigations since...